### ISSUE TITLE
Cannot access backing file 'crc.qcow2'; Permission denied

### ISSUE THREAD

**Issue by rscher** [code-ready/crc#1776]
CRC v 1.20 doesn't  start, fails with error:
```
 (crc) Failed to start: virError(Code=38, Domain=18, Message='Cannot access backing file '/home/crcuser/.crc/cache/crc_libvirt_4.6.6/crc.qcow2' of storage file '/home/crcuser/.crc/machines/crc/crc.qcow2' (as uid:107, gid:107): Permission denied')
```
when  reverting back to  crc v1.17 , works  fine.

  * OS: Linux 
  * Hypervisor: KVM 
  * Did you run `crc setup` before starting it (Yes/No)? yes

## CRC version
```bash
# Put the output of `crc version`
[crcuser@zzxr7601 ~]$ crc version
CodeReady Containers version: 1.20.0+ef3f80d
OpenShift version: 4.6.6 (embedded in executable)
```
  
## CRC status
```bash
# Put the output of `crc status`
[crcuser@zzxr7601 ~]$ crc status
CRC VM:          Stopped
OpenShift:       Stopped
Disk Usage:      0B of 0B (Inside the CRC VM)
Cache Usage:     13.69GB
Cache Directory: /home/crcuser/.crc/cache


## CRC config
```bash
# Put the output of `crc config view`
[crcuser@zzxr7601 ~]$ crc config view
- cpus                                  : 16
- memory                                : 24576
- pull-secret-file                      : /home/crcuser/crc-pull-secret.txt

## Host Operating System
```bash
"Red Hat Enterprise Linux Server"
"VERSION="7.9 (Maipo)"
```

### Steps to reproduce

  1. crc setup
  2. crc start
  3. 
  4. 

### Expected
crc to start and running

### Actual
```
(crc) Failed to start: virError(Code=38, Domain=18, Message='Cannot access backing file '/home/crcuser/.crc/cache/crc_libvirt_4.6.6/crc.qcow2' of storage file '/home/crcuser/.crc/machines/crc/crc.qcow2' (as uid:107, gid:107): Permission denied')
```

### Logs
```
[crcuser@zzxr7601 ~]$ crc start
Changes to configuration property 'cpus' are only applied when the CRC instance is started.
If you already have a running CRC instance, then for this configuration change to take effect, stop the CRC instance with 'crc stop' and restart it with 'crc start'.
Changes to configuration property 'memory' are only applied when the CRC instance is started.
If you already have a running CRC instance, then for this configuration change to take effect, stop the CRC instance with 'crc stop' and restart it with 'crc start'.
Successfully configured pull-secret-file to /home/crcuser/crc-pull-secret.txt
DEBU CodeReady Containers version: 1.20.0+ef3f80d
DEBU OpenShift version: 4.6.6 (embedded in executable)
DEBU Couldn't set key CPE_NAME, no corresponding struct field found
DEBU Couldn't set key , no corresponding struct field found
DEBU Couldn't set key REDHAT_BUGZILLA_PRODUCT, no corresponding struct field found
DEBU Couldn't set key REDHAT_BUGZILLA_PRODUCT_VERSION, no corresponding struct field found
DEBU Couldn't set key REDHAT_SUPPORT_PRODUCT, no corresponding struct field found
DEBU Couldn't set key REDHAT_SUPPORT_PRODUCT_VERSION, no corresponding struct field found
INFO Checking if podman remote executable is cached
DEBU Currently podman remote is not supported
INFO Checking if goodhosts executable is cached
DEBU goodhost executable is not cached
INFO Caching goodhosts executable
DEBU Trying to extract goodhosts from crc executable
DEBU Extracting embedded 'goodhosts-cli-linux-amd64.tar.xz' from /usr/bin/crc to /tmp/crc211482787/goodhosts-cli-linux-amd64.tar.xz
DEBU Uncompressing /tmp/crc211482787/goodhosts-cli-linux-amd64.tar.xz to /tmp/crc211482787
DEBU Copying '/tmp/crc211482787/goodhosts' to '/home/crcuser/.crc/bin/goodhosts'
DEBU goodhost executable cached
DEBU Making /home/crcuser/.crc/bin/goodhosts suid
INFO Will use root access: change ownership of /home/crcuser/.crc/bin/goodhosts
DEBU Running '/usr/bin/sudo chown root /home/crcuser/.crc/bin/goodhosts'
INFO Will use root access: set suid for /home/crcuser/.crc/bin/goodhosts
DEBU Running '/usr/bin/sudo chmod u+s,g+x /home/crcuser/.crc/bin/goodhosts'
INFO Checking if CRC bundle is extracted in '$HOME/.crc'
DEBU stat /home/crcuser/.crc/cache/crc_libvirt_4.6.6.crcbundle: no such file or directory
INFO Extracting bundle from the CRC executable
DEBU Error changing /home/crcuser/.crc/cache permissions to 0775
DEBU Extracting embedded 'crc_libvirt_4.6.6.crcbundle' from /usr/bin/crc to /home/crcuser/.crc/cache/crc_libvirt_4.6.6.crcbundle
DEBU Uncompressing /home/crcuser/.crc/cache/crc_libvirt_4.6.6.crcbundle to /home/crcuser/.crc/cache
crc.qcow2: 10.16 GiB / 10.16 GiB [------------------------------------------------------------------------------------------] 100.00%
INFO Checking minimum RAM requirements
DEBU Total memory of system is 33734914048 bytes
INFO Checking if running as non-root
INFO Checking if Virtualization is enabled
DEBU Checking if the vmx/svm flags are present in /proc/cpuinfo
DEBU CPU virtualization flags are good
INFO Checking if KVM is enabled
DEBU Checking if /dev/kvm exists
DEBU /dev/kvm was found
INFO Checking if libvirt is installed
DEBU Checking if 'virsh' is available
DEBU 'virsh' was found in /usr/bin/virsh
DEBU Checking 'virsh capabilities' for libvirtd/qemu availability
DEBU Running 'virsh capabilities'
DEBU Found x86_64 hypervisor with 'hvm' capabilities
INFO Checking if user is part of libvirt group
DEBU Checking if current user is part of the libvirt group
DEBU Running '/usr/bin/groups crcuser'
DEBU Current user is already in the libvirt group
INFO Checking if libvirt daemon is running
DEBU Checking if libvirtd service is running
DEBU Running 'systemctl status virtqemud.socket'
DEBU Command failed: exit status 4
DEBU stdout:
DEBU stderr: Unit virtqemud.socket could not be found.
DEBU virtqemud.socket is neither running nor listening
DEBU Running 'systemctl status libvirtd.socket'
DEBU Command failed: exit status 4
DEBU stdout:
DEBU stderr: Unit libvirtd.socket could not be found.
DEBU libvirtd.socket is neither running nor listening
DEBU Running 'systemctl status virtqemud.service'
DEBU Command failed: exit status 4
DEBU stdout:
DEBU stderr: Unit virtqemud.service could not be found.
DEBU virtqemud.service is neither running nor listening
DEBU Running 'systemctl status libvirtd.service'
DEBU libvirtd.service is running
INFO Checking if a supported libvirt version is installed
DEBU Checking if libvirt version is >=3.4.0
DEBU Running 'virsh -v'
INFO Checking if crc-driver-libvirt is installed
DEBU Checking if crc-driver-libvirt is installed
DEBU crc-driver-libvirt executable is not cached
INFO Installing crc-driver-libvirt
DEBU Installing crc-driver-libvirt
DEBU Trying to extract crc-driver-libvirt from crc executable
DEBU Extracting embedded 'crc-driver-libvirt' from /usr/bin/crc to /tmp/crc628423078/crc-driver-libvirt
DEBU Copying '/tmp/crc628423078/crc-driver-libvirt' to '/home/crcuser/.crc/bin/crc-driver-libvirt'
DEBU crc-driver-libvirt is installed in /home/crcuser/.crc/bin
INFO Checking if systemd-networkd is running
DEBU Checking if systemd-networkd.service is running
DEBU Running 'systemctl status systemd-networkd.service'
DEBU Command failed: exit status 4
DEBU stdout:
DEBU stderr: Unit systemd-networkd.service could not be found.
DEBU systemd-networkd.service is not running
INFO Checking if NetworkManager is installed
DEBU Checking if 'nmcli' is available
DEBU 'nmcli' was found in /usr/bin/nmcli
INFO Checking if NetworkManager service is running
DEBU Checking if NetworkManager.service is running
DEBU Running 'systemctl status NetworkManager.service'
DEBU NetworkManager.service is already running
INFO Checking if /etc/NetworkManager/conf.d/crc-nm-dnsmasq.conf exists
DEBU Checking NetworkManager configuration
DEBU File not found: /etc/NetworkManager/conf.d/crc-nm-dnsmasq.conf: stat /etc/NetworkManager/conf.d/crc-nm-dnsmasq.conf: no such file or directory
INFO Writing Network Manager config for crc
DEBU Fixing NetworkManager configuration
INFO Will use root access: write NetworkManager configuration to /etc/NetworkManager/conf.d/crc-nm-dnsmasq.conf
INFO Will use root access: Changing permission for /etc/NetworkManager/conf.d/crc-nm-dnsmasq.conf to 644
DEBU Running '/usr/bin/sudo chmod 644 /etc/NetworkManager/conf.d/crc-nm-dnsmasq.conf'
DEBU Reloading NetworkManager
INFO Will use root access: executing systemctl daemon-reload command
DEBU Running '/usr/bin/sudo systemctl daemon-reload'
INFO Will use root access: executing systemctl reload NetworkManager
DEBU Running '/usr/bin/sudo systemctl reload NetworkManager'
DEBU NetworkManager configuration fixed
INFO Checking if /etc/NetworkManager/dnsmasq.d/crc.conf exists
DEBU Checking dnsmasq configuration
DEBU File not found: /etc/NetworkManager/dnsmasq.d/crc.conf: stat /etc/NetworkManager/dnsmasq.d/crc.conf: no such file or directory
INFO Writing dnsmasq config for crc
DEBU Fixing dnsmasq configuration
INFO Will use root access: write NetworkManager configuration to /etc/NetworkManager/dnsmasq.d/crc.conf
INFO Will use root access: Changing permission for /etc/NetworkManager/dnsmasq.d/crc.conf to 644
DEBU Running '/usr/bin/sudo chmod 644 /etc/NetworkManager/dnsmasq.d/crc.conf'
DEBU Reloading NetworkManager
INFO Will use root access: executing systemctl daemon-reload command
DEBU Running '/usr/bin/sudo systemctl daemon-reload'
INFO Will use root access: executing systemctl reload NetworkManager
DEBU Running '/usr/bin/sudo systemctl reload NetworkManager'
DEBU dnsmasq configuration fixed
INFO Checking if libvirt 'crc' network is available
DEBU Checking if libvirt 'crc' network exists
DEBU Running 'virsh --connect qemu:///system net-info crc'
DEBU Command failed: exit status 1
DEBU stdout:
DEBU stderr: error: failed to get network 'crc'
error: Network not found: no network with matching name 'crc'
DEBU Libvirt network crc not found
INFO Setting up libvirt 'crc' network
DEBU Creating libvirt 'crc' network
DEBU Running 'virsh --connect qemu:///system net-destroy crc'
DEBU Command failed: exit status 1
DEBU stdout:
DEBU stderr: error: failed to get network 'crc'
error: Network not found: no network with matching name 'crc'
DEBU Running 'virsh --connect qemu:///system net-undefine crc'
DEBU Command failed: exit status 1
DEBU stdout:
DEBU stderr: error: failed to get network 'crc'
error: Network not found: no network with matching name 'crc'
DEBU libvirt 'crc' network created
INFO Checking if libvirt 'crc' network is active
DEBU Checking if libvirt 'crc' network is active
DEBU Running 'virsh --connect qemu:///system net-info crc'
DEBU Libvirt crc network is not active
INFO Starting libvirt 'crc' network
DEBU Starting libvirt 'crc' network
DEBU Running 'virsh --connect qemu:///system net-start crc'
DEBU Running 'virsh --connect qemu:///system net-autostart crc'
DEBU libvirt 'crc' network started
Setup is complete, you can now run 'crc start' to start the OpenShift cluster
DEBU CodeReady Containers version: 1.20.0+ef3f80d
DEBU OpenShift version: 4.6.6 (embedded in executable)
DEBU Total memory of system is 33734914048 bytes
DEBU Couldn't set key CPE_NAME, no corresponding struct field found
DEBU Couldn't set key , no corresponding struct field found
DEBU Couldn't set key REDHAT_BUGZILLA_PRODUCT, no corresponding struct field found
DEBU Couldn't set key REDHAT_BUGZILLA_PRODUCT_VERSION, no corresponding struct field found
DEBU Couldn't set key REDHAT_SUPPORT_PRODUCT, no corresponding struct field found
DEBU Couldn't set key REDHAT_SUPPORT_PRODUCT_VERSION, no corresponding struct field found
INFO Checking if podman remote executable is cached
DEBU Currently podman remote is not supported
INFO Checking if goodhosts executable is cached
DEBU goodhost executable already cached
INFO Checking minimum RAM requirements
DEBU Total memory of system is 33734914048 bytes
INFO Checking if running as non-root
INFO Checking if Virtualization is enabled
DEBU Checking if the vmx/svm flags are present in /proc/cpuinfo
DEBU CPU virtualization flags are good
INFO Checking if KVM is enabled
DEBU Checking if /dev/kvm exists
DEBU /dev/kvm was found
INFO Checking if libvirt is installed
DEBU Checking if 'virsh' is available
DEBU 'virsh' was found in /usr/bin/virsh
DEBU Checking 'virsh capabilities' for libvirtd/qemu availability
DEBU Running 'virsh capabilities'
DEBU Found x86_64 hypervisor with 'hvm' capabilities
INFO Checking if user is part of libvirt group
DEBU Checking if current user is part of the libvirt group
DEBU Running '/usr/bin/groups crcuser'
DEBU Current user is already in the libvirt group
INFO Checking if libvirt daemon is running
DEBU Checking if libvirtd service is running
DEBU Running 'systemctl status virtqemud.socket'
DEBU Command failed: exit status 4
DEBU stdout:
DEBU stderr: Unit virtqemud.socket could not be found.
DEBU virtqemud.socket is neither running nor listening
DEBU Running 'systemctl status libvirtd.socket'
DEBU Command failed: exit status 4
DEBU stdout:
DEBU stderr: Unit libvirtd.socket could not be found.
DEBU libvirtd.socket is neither running nor listening
DEBU Running 'systemctl status virtqemud.service'
DEBU Command failed: exit status 4
DEBU stdout:
DEBU stderr: Unit virtqemud.service could not be found.
DEBU virtqemud.service is neither running nor listening
DEBU Running 'systemctl status libvirtd.service'
DEBU libvirtd.service is running
INFO Checking if a supported libvirt version is installed
DEBU Checking if libvirt version is >=3.4.0
DEBU Running 'virsh -v'
INFO Checking if crc-driver-libvirt is installed
DEBU Checking if crc-driver-libvirt is installed
DEBU Running '/home/crcuser/.crc/bin/crc-driver-libvirt version'
DEBU Found crc-driver-libvirt version 0.12.13
DEBU crc-driver-libvirt is already installed
INFO Checking if systemd-networkd is running
DEBU Checking if systemd-networkd.service is running
DEBU Running 'systemctl status systemd-networkd.service'
DEBU Command failed: exit status 4
DEBU stdout:
DEBU stderr: Unit systemd-networkd.service could not be found.
DEBU systemd-networkd.service is not running
INFO Checking if NetworkManager is installed
DEBU Checking if 'nmcli' is available
DEBU 'nmcli' was found in /usr/bin/nmcli
INFO Checking if NetworkManager service is running
DEBU Checking if NetworkManager.service is running
DEBU Running 'systemctl status NetworkManager.service'
DEBU NetworkManager.service is already running
INFO Checking if /etc/NetworkManager/conf.d/crc-nm-dnsmasq.conf exists
DEBU Checking NetworkManager configuration
DEBU NetworkManager configuration is good
INFO Checking if /etc/NetworkManager/dnsmasq.d/crc.conf exists
DEBU Checking dnsmasq configuration
DEBU dnsmasq configuration is good
INFO Checking if libvirt 'crc' network is available
DEBU Checking if libvirt 'crc' network exists
DEBU Running 'virsh --connect qemu:///system net-info crc'
DEBU Checking if libvirt 'crc' definition is up to date
DEBU Running 'virsh --connect qemu:///system net-dumpxml --inactive crc'
DEBU libvirt 'crc' network has the expected value
INFO Checking if libvirt 'crc' network is active
DEBU Checking if libvirt 'crc' network is active
DEBU Running 'virsh --connect qemu:///system net-info crc'
DEBU libvirt 'crc' network is already active
Checking file: /home/crcuser/.crc/machines/crc/.crc-exist
INFO Loading bundle: crc_libvirt_4.6.6.crcbundle ...
INFO Verifying bundle crc_libvirt_4.6.6.crcbundle ...
INFO Creating CodeReady Containers VM for OpenShift 4.6.6...
Found binary path at /home/crcuser/.crc/bin/crc-driver-libvirt
Launching plugin server for driver libvirt
Plugin server listening at address 127.0.0.1:41546
() Calling .GetVersion
Using API Version  1
() Calling .SetConfigRaw
() Calling .GetMachineName
(crc) Calling .GetMachineName
(crc) Calling .DriverName
Running pre-create checks...
(crc) Calling .PreCreateCheck
(crc) DBG | About to check libvirt version
(crc) DBG | Validating network
(crc) DBG | Validating storage pool
(crc) Calling .GetConfigRaw
Creating machine...
(crc) Calling .Create
(crc) DBG | Preparing /home/crcuser/.crc/machines/crc/crc.qcow2 for machine use
(crc) DBG | image creation took 61.682815ms
(crc) DBG | Verifying executable bit set on /home/crcuser/.crc/machines/crc
(crc) DBG | Setting executable bit set on /home/crcuser/.crc/machines/crc
(crc) DBG | Verifying executable bit set on /home/crcuser/.crc/machines
(crc) DBG | Setting executable bit set on /home/crcuser/.crc/machines
(crc) DBG | Verifying executable bit set on /home/crcuser/.crc
(crc) DBG | Setting executable bit set on /home/crcuser/.crc
(crc) DBG | Verifying executable bit set on /home/crcuser
(crc) DBG | Verifying executable bit set on /home
(crc) DBG | Defining VM...
(crc) DBG | Found x86_64 hypervisor with 'hvm' capabilities
(crc) DBG | No q35 machine type
(crc) DBG | Adding the file: /home/crcuser/.crc/machines/crc/.crc-exist
(crc) DBG | disk image capacity is already 33285996544 bytes
(crc) DBG | Starting VM crc
(crc) DBG | Validating network
(crc) DBG | Validating storage pool
(crc) Failed to start: virError(Code=38, Domain=18, Message='Cannot access backing file '/home/crcuser/.crc/cache/crc_libvirt_4.6.6/crc.qcow2' of storage file '/home/crcuser/.crc/machines/crc/crc.qcow2' (as uid:107, gid:107): Permission denied')
Making call to close driver server
(crc) Calling .Close
Successfully made call to close driver server
Making call to close connection to plugin binary
Error creating machine: Error creating the VM: Error creating machine: Error in driver during machine creation: virError(Code=38, Domain=18, Message='Cannot access backing file '/home/crcuser/.crc/cache/crc_libvirt_4.6.6/crc.qcow2' of storage file '/home/crcuser/.crc/machines/crc/crc.qcow2' (as uid:107, gid:107): Permission denied')
```

--
Notes:
 I tried all the suggestions  from those who had same issue (not reported on crc v1.20), and none of them worked.
can you advise ?
thanks,
Russ

**Comment by gbraad** [comment]
It is clearly a permission problem. Can you share more details about the user? like a partial view of `/etc/passwd' and '/etc/groups', such as 'libvirt' and 'crcuser'. Also, what about permissions of `/home/crcuser/.crc/machines/crc` ? Who owns the files and has access to the folder?

**Comment by praveenkumar** [comment]
Also can you use `ls -lZ /home/crcuser/.crc/machines/crc` so that we get the selinux context on those files/directory.

**Comment by rscher** [comment]
hey guys, apprec any help to get this resolved ...  this host used to work with previous versions of crc,
not sure what has changed ..

# /etc/passwd
crcuser:x:1004:1004:crcuser:/home/crcuser:/bin/bash
qemu:x:107:107:qemu user:/:/sbin/nologin

# /etc/group
root:x:0:
bin:x:1:
crcuser:x:1004:
libvirt:x:986:crcuser
kvm:x:36:qemu
qemu:x:107:

[crcuser@zzxr7601 binaries]$ ls -lZ /home/crcuser/.crc/cache/
drwx------ crcuser crcuser ?                                crc_libvirt_4.6.6
-rw------- crcuser crcuser ?                                crc_libvirt_4.6.6.crcbundle
[crcuser@zzxr7601 binaries]$ ls -lZ  /home/crcuser/.crc/machines/crc
-rw------- crcuser crcuser ?                                config.json
-rw------- crcuser crcuser ?                                crc.qcow2
[crcuser@zzxr7601 binaries]$ ls -lZ /home/crcuser/.crc/cache/crc_libvirt_4.6.6
-rw------- crcuser crcuser ?                                crc-bundle-info.json
-rw------- crcuser crcuser ?                                crc.qcow2
-r-------- crcuser crcuser ?                                id_rsa_crc
-rw------- crcuser crcuser ?                                kubeadmin-password
-rw------- crcuser crcuser ?                                kubeconfig
-rwx------ crcuser crcuser ?                                oc


Checking file: /home/crcuser/.crc/machines/crc/.crc-exist
INFO Loading bundle: crc_libvirt_4.6.6.crcbundle ... 
INFO Verifying bundle crc_libvirt_4.6.6.crcbundle ... 
INFO Creating CodeReady Containers VM for OpenShift 4.6.6... 
Found binary path at /home/crcuser/.crc/bin/crc-driver-libvirt
Launching plugin server for driver libvirt
Plugin server listening at address 127.0.0.1:38368
() Calling .GetVersion
Using API Version  1
() Calling .SetConfigRaw
() Calling .GetMachineName
(crc) Calling .GetMachineName
(crc) Calling .DriverName
Running pre-create checks...
(crc) Calling .PreCreateCheck
(crc) DBG | About to check libvirt version
(crc) DBG | Validating network
(crc) DBG | Validating storage pool
(crc) Calling .GetConfigRaw
Creating machine...
(crc) Calling .Create
(crc) DBG | Preparing /home/crcuser/.crc/machines/crc/crc.qcow2 for machine use
(crc) DBG | image creation took 106.661112ms
(crc) DBG | Verifying executable bit set on /home/crcuser/.crc/machines/crc
(crc) DBG | Setting executable bit set on /home/crcuser/.crc/machines/crc
(crc) DBG | Verifying executable bit set on /home/crcuser/.crc/machines
(crc) DBG | Setting executable bit set on /home/crcuser/.crc/machines
(crc) DBG | Verifying executable bit set on /home/crcuser/.crc
(crc) DBG | Setting executable bit set on /home/crcuser/.crc
(crc) DBG | Verifying executable bit set on /home/crcuser
(crc) DBG | Verifying executable bit set on /home
(crc) DBG | Defining VM...
(crc) DBG | Found x86_64 hypervisor with 'hvm' capabilities
(crc) DBG | No q35 machine type
(crc) DBG | Adding the file: /home/crcuser/.crc/machines/crc/.crc-exist
(crc) DBG | disk image capacity is already 33285996544 bytes
(crc) DBG | Starting VM crc
(crc) DBG | Validating network
(crc) DBG | Validating storage pool
(crc) Failed to start: virError(Code=38, Domain=18, Message='Cannot access backing file '/home/crcuser/.crc/cache/crc_libvirt_4.6.6/crc.qcow2' of storage file '/home/crcuser/.crc/machines/crc/crc.qcow2' (as uid:107, gid:107): Permission denied')
Making call to close driver server
(crc) Calling .Close
Successfully made call to close driver server
Making call to close connection to plugin binary
Error creating machine: Error creating the VM: Error creating machine: Error in driver during machine creation: virError(Code=38, Domain=18, Message='Cannot access backing file '/home/crcuser/.crc/cache/crc_libvirt_4.6.6/crc.qcow2' of storage file '/home/crcuser/.crc/machines/crc/crc.qcow2' (as uid:107, gid:107): Permission denied')

**Comment by cfergeau** [comment]
Can you try to move `~/.crc` away and run again `crc setup` / `crc start`? Or did you try that already?

**Comment by gbraad** [comment]
Duplicate of https://github.com/code-ready/crc/issues/1578
Please try a newer version of CRC, like v1.20

**Comment by rscher** [comment]
> Can you try to move `~/.crc` away and run again `crc setup` / `crc start`? Or did you try that already?

Yes, have done that.
I ran this delete crc script as well before 'crc start',  didnt resolve it-

[crcuser@zzxr7601 ~]# cat delete_crc.sh
#!/bin/bash
crc  -f delete
crc cleanup
virsh -c qemu:///system destroy crc
virsh -c qemu:///system undefine crc
rm -rf ~/.crc

I originally thought it was caused by this host's permissions,
but after reading several others postings with exact error on other crc/libvirt 
versions, opened this issue. 

Running latest version of libvirt..

[crcuser@zzxr7601 binaries]$ sudo yum info libvirt
Loaded plugins: langpacks, product-id, search-disabled-repos, subscription-manager
Installed Packages
Name        : libvirt
Arch        : x86_64
Version     : 4.5.0
Release     : 36.el7_9.3

Yet issue persists ...
(crc) DBG | Validating storage pool
(crc) Failed to start: virError(Code=38, Domain=18, Message='Cannot access backing file '/home/crcuser/.crc/cache/crc_libvirt_4.6.6/crc.qcow2' of storage file '/home/crcuser/.crc/machines/crc/crc.qcow2' (as uid:107, gid:107): Permission denied')

**Comment by cfergeau** [comment]
> but after reading several others postings with exact error on other crc/libvirt versions, opened this issue.

Where did you read these? Only ubuntu issues come to mind, and the q35 rhel7 regression which gave a different error message, and was fixed in 1.20.

**Comment by rscher** [comment]
looks like this is similar  issue I am getting ..
https://github.com/code-ready/crc/pull/1571/commits/efec5eabb7e9ed4d98d0723817a1456a4431540c

in the output above in logs, I see this error when extracting bundle from the CRC executable
**DEBU Error changing /home/crcuser/.crc/cache permissions to 0775**

can you advise on a workaround to get past this ?

**Comment by cfergeau** [comment]
Not sure it matters, but `-rw------- crcuser crcuser ?                                crc.qcow2` is unexpected, it's `-rw-r--r--` on my test machine, and has `unconfined_u:object_r:user_home_t:s0` selinux label. When the VM starts, the file is changed to be owned by `qemu:qemu` and the selinux label becomes `system_u:object_r:svirt_image_t:s0:c830,c1019`

**Comment by gbraad** [comment]
> Is selinux disabled on your machine? 

I guess, as `ls -alZ` did not return any selinux labels

**Comment by rscher** [comment]
yes , selinux is disabled

**Comment by rscher** [comment]
got past this error finally, selinux is not required.
I was getting this error during a 'crc start' when it was extracting bundle from the CRC executable:
    _DEBU Error changing /home/crcuser/.crc/cache permissions to 0775_

I manually changed the permissions after 'crc setup' and before 'crc start'
I used 0755 instead of 0775.

$ crc setup
$  chmod -R 0755 ~/.crc/cache
$  crc start -p ~/crc-pull-secret.txt  -m 24576 -c 16 --disable-update-check 
--
$  ll ~/.crc/machines/crc/
total 3384996
**-rw------- 1 qemu    qemu    2415001600 Dec 18 00:39 crc.qcow2**
-rw------- 1 crcuser crcuser      15215 Dec 17 23:53 kubeconfig
-rw------- 1 crcuser crcuser       1679 Dec 17 23:53 id_rsa
-rw------- 1 crcuser crcuser        381 Dec 17 23:53 id_rsa.pub
-rw------- 1 crcuser crcuser       1593 Dec 17 23:53 config.json

$ crc status
CRC VM:          Running
OpenShift:       Running (v4.6.6)
Disk Usage:      10.98GB of 32.72GB (Inside the CRC VM)
Cache Usage:     13.69GB

thanks for all your help, closing issue
-Russ

**Comment by MrSchoolcraft** [comment]
using ubuntu 20.04 (pop_os) I get a similar issue now with 

> INFO Verifying bundle crc_libvirt_4.6.6.crcbundle ... 
INFO Creating CodeReady Containers VM for OpenShift 4.6.6... 
Error creating machine: Error creating the VM: Error creating machine: Error in driver during machine creation: virError(Code=1, Domain=10, Message='internal error: process exited while connecting to monitor: 2020-12-27T22:11:36.542320Z qemu-system-x86_64: -blockdev {"driver":"file","filename":"/home/USER/.crc/cache/crc_libvirt_4.6.6/crc.qcow2","aio":"threads","node-name":"libvirt-2-storage","auto-read-only":true,"discard":"unmap"}: Could not open '/home/USER/.crc/cache/crc_libvirt_4.6.6/crc.qcow2': Permission denied')

**Comment by guillaumerose** [comment]
@MrSchoolcraft can you post your `/etc/os-release` file ? Just to check the value of ID and ID_LIKE. Thanks.

**Comment by chanchalbose** [comment]
crc working fine on linuxmint for last two years.
Tried to upgrade to version 1.20. It's now broken.
I get this output:
```
$ crc start -p ~/pull-secret.txt -m 24576 -c 4 --disable-update-check
WARN[0000] distribution-specific preflight checks are not implemented for linuxmint 
WARN distribution-specific preflight checks are not implemented for linuxmint 
INFO Checking if podman remote executable is cached 
INFO Checking if goodhosts executable is cached   
INFO Checking minimum RAM requirements            
INFO Checking if running as non-root              
INFO Checking if Virtualization is enabled        
INFO Checking if KVM is enabled                   
INFO Checking if libvirt is installed             
INFO Checking if user is part of libvirt group    
INFO Checking if libvirt daemon is running        
INFO Checking if a supported libvirt version is installed 
INFO Checking if crc-driver-libvirt is installed  
INFO Checking if libvirt 'crc' network is available 
INFO Checking if libvirt 'crc' network is active  
INFO Checking if NetworkManager is installed      
INFO Checking if NetworkManager service is running 
INFO Checking if /etc/NetworkManager/conf.d/crc-nm-dnsmasq.conf exists 
INFO Checking if /etc/NetworkManager/dnsmasq.d/crc.conf exists 
INFO Extracting bundle: crc_libvirt_4.6.3.crcbundle ... 
crc.qcow2: 11.57 GiB / 11.57 GiB [-------------------------------------] 100.00%
INFO Checking size of the disk image /home/chanchal/.crc/cache/crc_libvirt_4.6.3/crc.qcow2 ... 
INFO Creating CodeReady Containers VM for OpenShift 4.6.3... 
Error creating machine: Error creating the VM: Error creating machine: Error in driver during machine creation: virError(Code=1, Domain=10, Message='internal error: process exited while connecting to monitor: 2021-01-12T03:31:06.347401Z qemu-system-x86_64: -blockdev {"driver":"file","filename":"/home/chanchal/.crc/cache/crc_libvirt_4.6.3/crc.qcow2","aio":"threads","node-name":"libvirt-2-storage","auto-read-only":true,"discard":"unmap"}: Could not open '/home/chanchal/.crc/cache/crc_libvirt_4.6.3/crc.qcow2': Permission denied')

$ ls -l /home/chanchal/.crc/cache/crc_libvirt_4.6.3/crc.qcow2
-rw-r--r-- 1 libvirt-qemu kvm 12427526144 Jan 12 09:01 /home/chanchal/.crc/cache/crc_libvirt_4.6.3/crc.qcow2****
```

**Comment by chanchalbose** [comment]
Problem solved. 
Got it working in Linux Mint 20, by adding the path to `/etc/apparmor.d/libvirt/TEMPLATE.qemu`

```
profile LIBVIRT_TEMPLATE flags=(attach_disconnected) {
  #include <abstractions/libvirt-qemu>
  /home/chanchal/.crc/cache/crc_libvirt_4.6.3/crc.qcow2 rk,
}
```

**Comment by sseide** [comment]
@chanchalbose  - thanks, this hint help to solve it to for me

CRC 1.32.0 / OCP 4.17.8
Linux Debian 12 bookworm

On Debian i had to modify two different profiles to got it working and reload he profiles afterwards:

`/etc/apparmor.d/libvirt/TEMPLATE.qemu` as you have written:
```
profile LIBVIRT_TEMPLATE flags=(attach_disconnected) {
#include <abstractions/libvirt-qemu>
@{HOME}/.crc/cache/*/crc.qcow2 rk,
}
```
and `/etc/apparmor.d/usr.lib.libvirt.virt-aa-helper` for file `/home/user/.crc/machines/crc/crc.qcow2` - here it was a bit more tricky as default rules for "virt-aa-helper" deny access to all hidden files in the user home directory:
```
audit deny @{HOME}/.* mrwkl,
audit deny @{HOME}/.*/ rw,
audit deny @{HOME}/.*/** mrwkl,
```
Therefor just adding an allow rule for the `crc.qcow2` file does not work. These deny rules must be changed to allow reading from `~/.crc/` directory...

@cfergeau ~@guillaumerose~ @gbraad 
I think it would be really helpful to have a note in the installation instructions for all Debian/Ubuntu based distributions to check current AppArmor profiles and system log messages for DENIED entries in case "crc start" does not work. For me it needed a long time to find the problem and this issue to solve the generic Access Denied error  message on startup.

**Comment by gbraad** [comment]
Debian uses Apparmor by default?

**Comment by sseide** [comment]
Hello - i'm not sure but i think package "apparmor" is not installed as default - at least it is not part of the "standard/important/required" Task selections. But this system i tried to setup CRC is installed a long time ago with many version updates in between. Therefor cannot say how/when package "apparmor" was installed.

But as soon as this package gets installed it is enabled. Kernel support is there with standard kernels. Nothing more to do.

So for documentation purpose a simple hint to check your kernel logs for AppArmor DENIED messages like  
```
audit: type=1400 audit(1707909202.515:74): apparmor="DENIED" operation="open" profile="virt-aa-helper" name="/home/..." pid=34643 comm="virt-aa-helper" requested_mask="r" denied_mask="r" fsuid=0 ouid=64055
```
would be enough to help fixing it. The error message from `crc start` might not be enough for most persons not that deep into system administration...

Thanks

**Comment by gbraad** [comment]
We will check this, as something similar might be the case for Suse. Although don't remember a similar report before.

A note in the wiki is for sure good start, as it an always be promoted if more widespread
